Brokerages/Zerodha/ZerodhaBrokerage.cs

```diff
@@ -0,0 +1,1079 @@
+                    list.Add(Symbol.Create(item.Value.Symbol, SecurityType.Equity, _market));
+                }
+            }
+            return list;
+        }
+
+        public IEnumerable<Slice> GetHistory(IEnumerable<Data.HistoryRequest> requests, DateTimeZone sliceTimeZone)
+        {
+
+            foreach (var request in requests)
+            {
+                if (request.Symbol.ID.SecurityType == SecurityType.Cfd || request.Symbol.ID.SecurityType == SecurityType.Crypto || request.Symbol.ID.SecurityType == SecurityType.Forex || request.Symbol.ID.SecurityType == SecurityType.Commodity)
+                {
+                    throw new ArgumentException("Zerodha does not support this security type: " + request.Symbol.ID.SecurityType);
+                }
+
+                if (request.StartTimeUtc >= request.EndTimeUtc)
+                {
+                    throw new ArgumentException("Invalid date range specified");
+                }
+
+                var start = request.StartTimeUtc.ConvertTo(DateTimeZone.Utc, TimeZones.Kolkata);
+                var end = request.EndTimeUtc.ConvertTo(DateTimeZone.Utc, TimeZones.Kolkata);
+
+                var history = Enumerable.Empty<Slice>();
+
+
+                switch (request.Resolution)
+                {
+                    case Resolution.Tick:
+                        history = GetHistoryTick(request.Symbol, start, end);
+                        break;
+
+                    case Resolution.Second:
+                        history = GetHistorySecond(request.Symbol, start, end);
+                        break;
+
+                    case Resolution.Minute:
+                        history = GetHistoryMinute(request.Symbol, start, end);
+                        break;
+
+                    case Resolution.Hour:
+                        history = GetHistoryHour(request.Symbol, start, end);
+                        break;
+
+                    case Resolution.Daily:
+                        history = GetHistoryDaily(request.Symbol, start, end);
+                        break;
+                }
+
+                foreach (var slice in history)
+                {
+                    yield return slice;
+                }
+            }
+        }
+
+        private IEnumerable<Slice> GetHistoryDaily(Symbol symbol, DateTime start, DateTime end)
+        {
+            Log.Debug("ZerodhaBrokerage.GetHistoryDaily();");
+            string scripSymbol = _symbolMapper.GetBrokerageSymbol(symbol);
+            var candles = _kite.GetHistoricalData(scripSymbol, start, end, "day");
+
+            if (!candles.Any())
+            {
+                OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Warning, "NoHistoricalData",
+                    $"Exchange returned no data for {symbol} on history request " +
+                    $"from {start:s} to {end:s}"));
+            }
+
+            if (candles == null)
+                return Enumerable.Empty<Slice>();
```

Comment:
This check should be moved before the previous `if` block,
also in the other `GetHistory` methods

Reply:
done!